### PR TITLE
[16.11] Pass evaluated AssemblyName during Roslyn workspace construction

### PR DESCRIPTION
**Pull request by drewnoakes**
Cherry-pick of #7216 for 16.11.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7295)

### CODE REVIEW

File: build/import/Packages.targets [hunk 1]
```diff
@@ -94,8 +94,8 @@
     <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Query"                             Version="16.10.337-pre" />
 
     <!-- Roslyn -->
-    <PackageReference Update="Microsoft.VisualStudio.LanguageServices"                                Version="3.10.0-2.21121.14" />
-    <PackageReference Update="Microsoft.CodeAnalysis"                                                 Version="3.10.0-2.21121.14" />
+    <PackageReference Update="Microsoft.VisualStudio.LanguageServices"                                Version="4.0.0-2.21302.13" />
+    <PackageReference Update="Microsoft.CodeAnalysis"                                                 Version="4.0.0-2.21302.13" />
```
Comment (drewnoakes) [inline build/import/Packages.targets:98]: @jasonmalinowski are these package versions appropriate for 16.11?

Comment (drewnoakes) [reply build/import/Packages.targets:98]: I suspect not, based on the build errors around package downgrades. For example:

> NU1605: Detected package downgrade: Microsoft.VisualStudio.Threading from 17.0.15-alpha to 16.10.53-alpha

Comment (jasonmalinowski) [reply build/import/Packages.targets:98]: No, we'll need to get you a 3.11 version. PR now open to create it: https://github.com/dotnet/roslyn/pull/53855

**Review by MiYanni — APPROVED** [review]

**Comment by drewnoakes** [comment]
Thank you for fixing the build issue @jmarolf ❤️

**Pull request merged**